PACKAGE = "rm_dart"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# HSV
gen.add("green_h_max", int_t, 0, "Green h_max", 77, 0, 180)
gen.add("green_h_min", int_t, 0, "Green h_min", 35, 0, 180)
gen.add("green_s_max", int_t, 0, "Green s_max", 255, 0, 255)
gen.add("green_s_min", int_t, 0, "Green s_min", 43, 0, 255)
gen.add("green_v_max", int_t, 0, "Green v_max", 255, 0, 255)
gen.add("green_v_min", int_t, 0, "Green v_min", 46, 0, 255)

draw_type = gen.enum([gen.const("DISABLE", int_t, 0, "DISABLE"),
                       gen.const("RAW", int_t, 1, "RAW"),
                       gen.const("BINARY", int_t, 2, "BINARY"),
                       gen.const("MORPHOLOGY", int_t, 3, "MORPHOLOGY"),
                       gen.const("TARGET", int_t, 4, "TARGET")], "type")

gen.add("draw_type", int_t, 0, "", 0, 0, 4, edit_method=draw_type)

exit(gen.generate(PACKAGE, "rm_dart", "Dart"))